semeio/jobs/rft/gendata_rft.py

```diff
@@ -24,24 +24,48 @@ def _write_gen_data_files(trajectory_df, fname):
             hold pressure data, but will be defaulted if not present. "is_active"
             should be a boolean column, and "inactive_info" should contain
             strings with information on why points are inactive.
-        fname (str): Filename to hold the values for the GEN_DATA keyword,
-            which are pressure values in this case, and used as a basename
-            for the two auxiliary files.
+        directory (str): Directory name, for where to dump files.
+        well (str): Name of well, to be used to construct filenames.
+        report_step (int): The RFT report step, used to construct the filename
     """  # noqa
-    _write_pressure(fname, trajectory_df)
-    _write_active(fname + "_active", trajectory_df)
-    _write_inactive_info(fname + "_inactive_info", trajectory_df)
 
-
-def _write_pressure(fname, trajectory_df):
+    _write_simdata(
+        os.path.join(directory, "RFT_{}_{}".format(well, report_step)),
+        "pressure",
+        trajectory_df,
+    )
+    if "swat" in trajectory_df:
```

Comment:
You are making this test both inside and outside `_write_simdata`. I get why, but is it needed? The flip side is that without this check here, there api is more predictable (instead of outputting a variable number of files).

Reply:
`_write_simdata` is now predictable (or it emits error messages), and `_write_gen_data_files` outputs a variable number of files, but always the pressure file.